lib/entities/artifact/CArtHandler.cpp

```diff
@@ -204,7 +204,23 @@ std::shared_ptr<CArtifact> CArtHandler::loadFromJson(const std::string & scope,
 	{
 		if (b.second.isNull())
 			continue;
-		auto bonus = JsonUtils::parseBonus(b.second, art->getBonusTextID(b.first));
+
+		std::shared_ptr<Bonus> bonus;
+		if(!b.second["spell"].isNull()) // TODO How to check "chargeBasedSpell" struct?
+		{
+			// Parse "chargeBasedSpell"
+			bonus = JsonUtils::parseChargeBasedSpell(b.second, [b, art](const SpellID & spellId)
+			{
+				if(!b.second["chargeCost"].isNull())
+					art->setChargeCostForSpell(spellId, b.second["chargeCost"].Float());
+				else
+					art->setChargeCostForSpell(spellId, 1);
+			});
+		}
```

Comment:
Might be better to use separate field.
Right now you have one field that is used for two rather different use-cases, even if they are somewhat similar internally. Most of this loop is "if" block for example.